CDCgov/prime-reportstream
# 7250 load sender transform valuesets from lookup tables

This PR makes changes to the handling of sender transform valuesets to allow extensibility and extends valuesets to support retrieving data from RS lookup tables.

Test Steps:
1. Add a sender transform that uses a lookup table to `default-sender-transform.yml`:
```
  - name: test-transform
    resource: 'Bundle.entry.resource.ofType(Patient).name'
    condition: 'Bundle.entry.resource.ofType(Patient).exists()'
    bundleProperty: '%resource.family'
    value: [ '%resource.family' ]
    valueSet:
      lookupTable:
        tableName: npi-lookup
        keyColumn: first_name
        valueColumn: last_name
```
2. Send a report to the `full-elr` pipeline that will have data that should be transformed, e.g. patient's last name appears in the list of first names in the `npi-lookup` table.
3. Check the sftp output of the report and verify the transform occurred.

## Changes
- Created `ValueSetCollection` interface, which replaces `SortedMap<String, String>` in `ConfigSchemaElement`. The interface describes function `getMappedValue(String)`, which returns the corresponding value for a given key.
- Created `inlineValueSet` class which acts as the default implementation of `ValueSetCollection`, duplicating the previous behavior (with a slight change to the config schema).
- Created `lookupValueSet` class which is an extended implementation of `ValueSetCollection` specific to RS that retrieves valueset data from a given lookup table.
- Updated `ConfigSchemaElement` and its child objects to utilize the `ValueSetCollection` interface. 
- Updated `ConfigSchemaReader` to attach the `lookupValueSet` class as a possible subtype when deserializing transform schemas.
- Updated all existing HL7 mapping files to match changes to the config schema.
- Moved logic to retrieve valueset value for a given key from `ConfigSchemaProcessor` to `ValueSetCollection`.
- Refactored schemas so that `value` and `valueSet` are nullable instead of defaulting to empty, and updated usage accordingly.

## Checklist

### Testing

## Linked Issues
- #7250
